crd/apis/network/v1/gkenetworkparams_types.go

```diff
@@ -0,0 +1,54 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GKENetworkParams represent GKE specific parameters for the network.
+type GKENetworkParams struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GKENetworkParamsSpec   `json:"spec,omitempty"`
+	Status GKENetworkParamsStatus `json:"status,omitempty"`
+}
+
+// SecondaryRanges represents ranges of network addresses.
+type SecondaryRanges struct {
+	// +kubebuilder:validation:MinItems:=1
+	RangeNames []string `json:"rangeNames"`
```

Comment:
why is it a list? I don't see the gcloud API allow multiple secondary ranges

Reply:
Eventually, when we want to support multi pod CIDR, it needs to be a list. This is future proofing.